partiql-planner/src/main/kotlin/org/partiql/planner/catalog/Identifier.kt

```diff
@@ -0,0 +1,173 @@
+package org.partiql.planner.catalog
+
+/**
+ * Represents an SQL identifier (possibly qualified).
+ *
+ * @property qualifier   If
+ * @property identifier
+ */
+public class Identifier private constructor(
+    private val qualifier: Array<Part>,
+    private val identifier: Part,
+) {
+
+    /**
+     * Returns the unqualified name part.
+     */
+    public fun getIdentifier(): Part = identifier
+
+    /**
+     * Returns the name's namespace.
+     */
+    public fun getQualifier(): Array<Part> = qualifier
+
+    /**
+     * Returns true if the namespace is non-empty.
+     */
+    public fun hasQualifier(): Boolean = qualifier.isNotEmpty()
+
+    /**
+     * Compares one identifier to another, possibly ignoring case.
+     */
+    public fun matches(other: Identifier, ignoreCase: Boolean = false): Boolean {
+        //
+        if (this.qualifier.size != other.qualifier.size) {
+            return false
+        }
+        // Compare identifier
+        if (ignoreCase && !(this.identifier.matches(other.identifier))) {
+            return false
```

Comment:
Beyond this, what is the purpose of `matches`? From what I'm gathering, `Name` is the resolved unix-style path of a DB object, whereas `Identifier` is unresolved. Why would you want to compare two identifiers? In the current implementation, we have `BindingPath.matches(ConnectorPath)` (AKA `unresolved.matches(resolved)`)

Reply:
Correct that the above won't actually ignore the case (because these are delimited), but consider if one or more of the identifiers is "regular" but the planner is in _case-sensitive_ mode.

The `ignoreCase = false` is an explicit way to **ignore** regular vs. delimited and just compare the text.

On the second point, what is the purpose of matches? I don't actually believe we need to compare two identifiers, but I do know the catalog resolution implementation will be different than bindingpath.matches(connectorpath) of today. I can remove the match logic later.